### PR TITLE
sfst 1.4.7d (new formula)

### PR DESCRIPTION
**Pull request by garfieldnate**
Add a new formula for SFST, which is used for working with finite state
transducers, particularly in morphological analysis.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by garfieldnate** [comment]
Don't close. I want it merged.

File: Formula/sfst.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class Sfst < Formula
+  desc "Toolbox for morphological analysers and other FST-based tools"
+  homepage "http://www.cis.uni-muenchen.de/~schmid/tools/SFST/"
+  url "http://www.cis.uni-muenchen.de/~schmid/tools/SFST/data/SFST-1.4.7d.tar.gz"
+  sha256 "5a13c6a45298197216a6299eb6cdf96595d2036572bb518b9e1c1893cb1a6d5f"
+  # dependencies are bison, flex and ncurses, but they are included in MacOS
```
Comment (DomT4) [inline Formula/sfst.rb]: Don't need this line.

File: Formula/sfst.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def install
+    cd "src"
```
Comment (DomT4) [inline Formula/sfst.rb]: `cd "src" do` is the preferred style here.

File: Formula/sfst.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    system "make"
+    system "make", "DESTDIR=#{prefix}/", "install"
+    system "make", "DESTDIR=#{share}/", "maninstall"
+  end
+
+  test do
+    # this simple example is from the manual.
+    # first compile a simple grammar
+    File.write("foo.fst", "Hello")
```
Comment (DomT4) [inline Formula/sfst.rb]: `(testpath/"foo.fst").write "Hello"`

File: Formula/sfst.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    system "#{bin}/fst-compiler", "foo.fst", "foo.a"
+    assert_true File.file?("foo.a")
```
Comment (DomT4) [inline Formula/sfst.rb]: `assert_predicate testpath/"foo.a", :exist?, "Foo.a should exist but does not!"` or something similar.

File: Formula/sfst.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    # then parse simple input using that grammar
+    require "open3"
```
Comment (DomT4) [inline Formula/sfst.rb]: `require`s should be declared at the top of the block.

**Comment by garfieldnate** [comment]
@DomT4 thanks for the feedback! I incorporated and pushed your changes.

File: Formula/sfst.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    cd "src" do
+      system "make"
+      system "make", "DESTDIR=#{prefix}/", "install"
+      system "make", "DESTDIR=#{share}/", "maninstall"
+    end
+  end
+
+  test do
+    require "open3"
+    # this simple example is from the manual.
```
Comment (DomT4) [inline Formula/sfst.rb]: You can probably remove these two comments, and leave an empty line between the `require` and the `(tespath` lines. I'm aware this is a fairly anal thing to care about 😅.

File: Formula/sfst.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    # first compile a simple grammar
+    (testpath/"foo.fst").write "Hello"
+    system "#{bin}/fst-compiler", "foo.fst", "foo.a"
+    assert_predicate testpath/"foo.a", :exist?, "Foo.a should exist but does not!"
+
+    # then parse simple input using that grammar
```
Comment (DomT4) [inline Formula/sfst.rb]: Same with this line. The block seems self-explanatory enough.

**Comment by garfieldnate** [comment]
@DomT4 done!

**Comment by DomT4** [comment]
@ilovezfs @commitay This could use one of you to look it over, when you have time.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/515cc83a9e2db4e3925a4714a0dcfc4998942598. Thank you for your contribution to Homebrew @garfieldnate; we appreciate it! :smiley_cat:

**Comment by garfieldnate** [comment]
🎉 Thanks!

**Pull request closed without merging**